### PR TITLE
feat: make LSP page multi chain compatible

### PR DESCRIPTION
**Pull request by amateima**

### CODE REVIEW

File: components/lsp/MintForm.tsx [hunk 1]
```diff
@@ -180,6 +188,14 @@ const MintForm: FC<Props> = ({
     userNeedsToApprove,
   ]);
 
+  if (web3Provider && !isUserConnectedToContractChain) {
+    return (
+      <SwitchWalletContainer>
+        <SwitchWalletLsp targetChainId={chainId} />
+      </SwitchWalletContainer>
```
Comment (daywiss) [inline components/lsp/MintForm.tsx:195]: can u add a screenshot of what this looks like?

File: components/LiveIndicator.tsx [hunk 2]
```diff
@@ -23,13 +23,14 @@ export const LiveIndicator: React.FC<LiveIndicatorProps> = ({
...
 const Wrapper = styled.div`
-  background-color: var(--gray-600);
+  background-color: var(--white);
   border-radius: 5px;
   padding: 5px 25px 5px 10px;
   text-transform: uppercase;
   display: flex;
   align-items: center;
   width: fit-content;
+  color: #272528;
```
Comment (Gamaranto) [inline components/LiveIndicator.tsx:33]: nit: we should use CSS variables instead of hardcoded values for colors, unless its a one off thing

File: components/LspHero.tsx [hunk 3]
```diff
@@ -63,7 +69,12 @@ export const LspHero: React.FC<Props> = ({
           </Button>
         </CardHead>
         {isConnected && <Account id="walletAccount">{account}</Account>}
-        {isConnected && (
+        {isConnected && !isUserConnectedToContractChain && (
```
Comment (Gamaranto) [inline components/LspHero.tsx:72]: nothing wrong with the condition here, but it could get a bit hard to follow with multiple of those in the JSX, we could define a variable that tells us exactly what the issue is like `hasToChangeChain` and it could be defined in the component body above, so it does not clutter the JSX

File: components/lsp/SwitchWalletLsp.styled.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  padding: 0.75rem 1rem;
+  display: flex;
+  align-items: center;
+  border-radius: 8px;
+  font-size: 1.125rem;
+  font-weight: 400;
+  font-family: inherit;
+  color: #272528;
+  background-color: #fbfafb;
+  border: 1px solid hsla(0, 0%, 77%, 1);
```
Comment (Gamaranto) [inline components/lsp/SwitchWalletLsp.styled.tsx:26]: same nit about the colors

File: components/lsp/SwitchWalletLsp.styled.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  line-height: 22px;
+  text-align: center;
+  color: #272528;
+`;
+
+export const SwitchButton = styled(BaseButton)`
+  width: 100%;
+  max-width: 370px;
+  margin: 0 auto;
+  padding: 0.75rem 1rem;
```
Comment (Gamaranto) [inline components/lsp/SwitchWalletLsp.styled.tsx:17]: Is this is using `rem` because we want the padding to scale with the user font-size? If there is no specific reason for using `rems` it would be nice to use `px` just to keep consistency with other parts of the app

File: components/lsp/SwitchWalletLsp.styled.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import styled from "@emotion/styled";
+import { BaseButton } from "..";
+
+export const Disclaimer = styled.p`
+  margin: 0 0 26px;
+  font-weight: 600;
+  font-size: 20px;
+  line-height: 22px;
```
Comment (Gamaranto) [inline components/lsp/SwitchWalletLsp.styled.tsx:8]: 2 nits, f
- font-size is set in px, which would not allow a user to rescale it if they changed the browser default font-size, a nice trick to set the font in px, but allowing the user to rescale is doing `font-size: ${20 / 16}rem`, which will be the rem value of 20px at 1rem = 16px, but will scale
- Setting line-height in px won't rescale it, I suggest we set it in absolute value like `line-height: ${22 / 20}` if we are targeting 22px of line-height at 1 rem = 16px

**Review by Gamaranto — COMMENTED** [review]
Some nits, otherwise looks really good!

File: components/lsp/MintForm.tsx (hunk 1 above)
Comment (amateima) [reply components/lsp/MintForm.tsx:195]: Sure
Screenshot 2022-02-07 at 15 20 10

File: pages/[chain]/[address].tsx [hunk 7]
```diff
@@ -327,10 +330,13 @@ const SynthPage: React.FC<Props> = ({ data, chainId, relatedSynths }) => {
...
             </Heading>
-            <Description>{data.address}</Description>
           </div>
-          <StyledLiveIndicator isLive={!isExpired} />
         </HeroContentWrapper>
+        <HeroChain
+          chainId={chainId}
+          contractAddress={data.address}
+          isExpired={isExpired}
+        />
```
[the remark this answers is not in the document]
Comment (amateima) [reply pages/[chain]/[address].tsx:339]: I posted a ss above an it contains this section as well 👍

File: components/lsp/SwitchWalletLsp.styled.tsx (hunk 5 above)
Comment (amateima) [reply components/lsp/SwitchWalletLsp.styled.tsx:17]: rem usage is intended here, yes

File: components/lsp/SwitchWalletLsp.styled.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+`;
+
+export const ChainIconContainer = styled.div`
+  width: 35px;
+  height: 35px;
+  margin-right: 1.25rem;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  background-color: #fff;
```
Comment (Gamaranto) [inline components/lsp/SwitchWalletLsp.styled.tsx:41]: nit: color is hardcoded

File: pages/[chain]/[address].tsx [hunk 9]
```diff
@@ -541,3 +583,87 @@ const TableWrapper = styled.section`
...
+`;
+
+export const HeroChainIconContainer = styled.div`
+  width: 25px;
+  height: 25px;
+  margin-right: 0.875rem;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  background-color: #fff;
```
Comment (Gamaranto) [inline pages/[chain]/[address].tsx:662]: nit: color

File: pages/[chain]/[address].tsx [hunk 10]
```diff
@@ -541,3 +583,87 @@ const TableWrapper = styled.section`
...
   }
 `;
+
+const HeroChainWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  padding: 0 12px;
+  margin: 25px 0 0;
+  background: rgba(255, 255, 255, 0.05);
+  border: 1px solid rgba(255, 255, 255, 0.25);
```
Comment (Gamaranto) [inline pages/[chain]/[address].tsx:593]: same as above, one good way of doing it is `hsla(${COLORS.white / 0.25)`

File: components/lsp/SwitchWalletLsp.tsx [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+  const chainName = useMemo(
+    () => capitalize(chainIdToNameLookup[targetChainId]),
+    [targetChainId]
+  );
+
+  const onClickSwitch = async () => {
+    try {
+      await switchToNetwork(targetChainId);
+    } catch (error) {
+      if ((error as any).code === 4902) {
```
[the remark this answers is not in the document]
Comment (amateima) [reply components/lsp/SwitchWalletLsp.tsx:38]: I don't think so. It would just be undefined
